pysb/simulator/kappa.py

```diff
@@ -0,0 +1,178 @@
+        points = len(self.tspan)
+        time = self.tspan[-1]
+        plot_period = time / (len(self.tspan) - 1)
+
+        if output_file_basename is None:
+            output_file_basename = 'tmpKappa_%s_' % self.model.name
+
+        base_directory = tempfile.mkdtemp(prefix=output_file_basename,
+                                          dir=output_dir)
+
+        base_filename = os.path.join(base_directory, self.model.name)
+        kappa_filename_pattern = base_filename + '_{}.ka'
+        out_filename_pattern = base_filename + '_{}_run{}.out'
+
+        base_args = ['-u', 'time', '-l', str(time), '-p', '%.5f' % plot_period]
+        if 'seed' in additional_args:
+            base_args.extend(['-seed', str(additional_args['seed'])])
+
+        kasim_path = pf.get_path('kasim')
+        n_param_sets = self.initials_length
+
+        gen = KappaGenerator(self.model, _exclude_ic_param=True)
+        file_data_base = gen.get_content()
+
+        # Kappa column names, for sanity check
+        kappa_col_names = tuple(['time'] +
+                                [o.name for o in self.model.observables])
+        tout = []
+        observable_traj = []
+        try:
+            for pset_idx in range(n_param_sets):
+                file_data = file_data_base + ''
+                for param, param_value in zip(self.model.parameters,
+                                              self.param_values[pset_idx]):
+                    file_data += "%var: '{}' {:e}\n".format(param.name,
+                                                            param_value)
+                file_data += '\n'
+                for cp, values in self.initials_dict.items():
+                    file_data += "%init: {} {}\n".format(
+                        values[pset_idx], gen.format_complexpattern(cp)
+                    )
+
+                # Generate the Kappa model code from the PySB model and write
+                # it to the Kappa file:
+                kappa_filename = kappa_filename_pattern.format(pset_idx)
+                with open(kappa_filename, 'w') as kappa_file:
+                    self._logger.debug('Kappa file contents:\n\n' + file_data)
+                    kappa_file.write(file_data)
+
+                for sim_rpt in range(n_runs):
+                    # Run Kappa
+                    out_filename = out_filename_pattern.format(pset_idx,
+                                                               sim_rpt)
+                    args = [kasim_path] + base_args + [
+                        '-i', kappa_filename, '-o', out_filename]
+
+                    # Run KaSim
+                    self._logger.debug('Running: {}'.format(' '.join(args)))
+                    p = subprocess.Popen(args,
+                                         stdout=subprocess.PIPE,
+                                         stderr=subprocess.PIPE,
+                                         cwd=base_directory)
+
+                    for line in iter(p.stdout.readline, b''):
```

Comment:
Does this do something different here than the shorter `for line in p.stdout` ?

Reply:
No, `for line in p.stdout` seems to work too. I'll switch to that.